src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -3966,3 +4114,120 @@ void DoorLockServer::DoorLockOnAutoRelockCallback(System::Layer *, void * callba
         ChipLogProgress(Zcl, "Door Auto relock timer expired. %s", "Already locked.");
     }
 }
+
+CHIP_ERROR DoorLockServer::ReadAliroExpeditedTransactionSupportedProtocolVersions(const ConcreteReadAttributePath & aPath,
+                                                                                  AttributeValueEncoder & aEncoder,
+                                                                                  Delegate * delegate)
+{
+    VerifyOrReturnError(delegate != nullptr, CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "Delegate is nullptr"));
+
+    return aEncoder.EncodeList([delegate](const auto & encoder) -> CHIP_ERROR {
+        for (uint8_t i = 0; true; i++)
+        {
+            uint8_t buffer[kProtocolVersionMaxSize];
+            MutableByteSpan protocolVersion(buffer);
+            auto err = delegate->GetAliroExpeditedTransactionSupportedProtocolVersionAtIndex(i, protocolVersion);
+            if (err == CHIP_ERROR_PROVIDER_LIST_EXHAUSTED)
```

Comment:
See, this CHIP_ERROR_PROVIDER_LIST_EXHAUSTED thing _needs to be documented in the API_.

Reply:
done.